generated-code/generate.py

```diff
@@ -0,0 +1,219 @@
+  cmdLineParser.add_argument('--executable_libxsmm', default='')
+  cmdLineParser.add_argument('--executable_pspamm', default='')
+  cmdLineParser.set_defaults(enable_premultiply_flux=False)
+  cmdLineArgs = cmdLineParser.parse_args()
+
+  # derive the compute platform
+  gpu_platforms = ['cuda', 'hip', 'hipsycl', 'acpp', 'oneapi']
+  targets = ['gpu', 'cpu'] if cmdLineArgs.device_backend in gpu_platforms else ['cpu']
+
+  if cmdLineArgs.device_backend == 'none':
+    arch = useArchitectureIdentifiedBy(cmdLineArgs.host_arch)
+  else:
+    arch = useArchitectureIdentifiedBy(cmdLineArgs.host_arch, cmdLineArgs.device_arch, cmdLineArgs.device_backend)
+
+  # pick up the gemm tools defined by the user
+  gemm_tool_list = cmdLineArgs.gemm_tools.replace(" ", "").split(",")
+  gemm_generators = []
+
+  for tool in gemm_tool_list:
+    if hasattr(gemm_configuration, tool):
+      specific_gemm_class = getattr(gemm_configuration, tool)
+      # take executable arguments, but only if they are not empty
+      if specific_gemm_class is gemm_configuration.LIBXSMM and cmdLineArgs.executable_libxsmm != '':
+        gemm_generators.append(specific_gemm_class(arch, cmdLineArgs.executable_libxsmm))
+      elif specific_gemm_class is gemm_configuration.PSpaMM and cmdLineArgs.executable_pspamm != '':
+        gemm_generators.append(specific_gemm_class(arch, cmdLineArgs.executable_pspamm))
+      else:
+        gemm_generators.append(specific_gemm_class(arch))
+    elif tool.strip().lower() != 'none':
+      print("YATETO::ERROR: unknown \"{}\" GEMM tool. "
+            "Please, refer to the documentation".format(tool))
+      sys.exit("failure")
+
+  cost_estimators = BoundingBoxCostEstimator
+  if 'gpu' in targets:
+    try:
+      chainforge_spec = importlib.util.find_spec('chainforge')
+      chainforge_spec.loader.load_module()
+      cost_estimators = FusedGemmsBoundingBoxCostEstimator
+    except:
+      print('WARNING: ChainForge was not found. Falling back to GemmForge.')
+
+  subfolders = []
+
+  equationsSpec = importlib.util.find_spec(f'kernels.equations.{cmdLineArgs.equations}')
+  try:
+    equations = equationsSpec.loader.load_module()
+  except:
+    raise RuntimeError('Could not find kernels for ' + cmdLineArgs.equations)
+
+  if cmdLineArgs.equations == 'anisotropic':
+    equation_class = equations.AnisotropicADERDG
+  elif cmdLineArgs.equations == 'elastic':
+    equation_class = equations.ElasticADERDG
+  elif cmdLineArgs.equations == 'viscoelastic':
+    equation_class = equations.ViscoelasticADERDG
+  elif cmdLineArgs.equations == 'viscoelastic2':
+    equation_class = equations.Viscoelastic2ADERDG
+  else:
+    equation_class = equations.PoroelasticADERDG
+
+  def generate_equation(subfolders, equation, order):
+    precision = 'double' if cmdLineArgs.host_arch[0] == 'd' else 'single'
+
+    if cmdLineArgs.memLayout == 'auto':
+      # TODO(Lukas) Don't hardcode this
```

Comment:
Does this ToDo still make any sense for us? I don't understand which hardcoding this comment refers to

Reply:
They still do as far as I'm aware: we need to rename the parameters (while that shouldn't happen)—that's what this TODO should refer to if I read it right.